core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1173,17 +1187,37 @@ public ByteSizeValue getTotalIndexingBufferBytes() {
      * @param loader loads the data into the cache if needed
      * @return the contents of the cache or the result of calling the loader
      */
-    private BytesReference cacheShardLevelResult(IndexShard shard, DirectoryReader reader, BytesReference cacheKey, Loader loader)
+    private Tuple<Boolean, BytesReference> cacheShardLevelResult(IndexShard shard, DirectoryReader reader, BytesReference cacheKey, Consumer<StreamOutput> loader)
             throws Exception {
-        IndexShardCacheEntity cacheEntity = new IndexShardCacheEntity(shard, loader);
-        return indicesRequestCache.getOrCompute(cacheEntity, reader, cacheKey);
+        IndexShardCacheEntity cacheEntity = new IndexShardCacheEntity(shard);
+        boolean[] loadedFromCache = new boolean[] { true };
```

Comment:
do we need to return this, I think we can rely on that the loader is called when we load the result again so the caller can use this to obtain that boolean...?